features/javascript/response.feature

```diff
@@ -0,0 +1,76 @@
+@javascript
+Feature: Response validation
+
+  Background:
+    Given you call:
+    """
+    const gavel = require('gavel');
+    """
+    And you define following HTTP response object:
+    """
+    const real = {
+      "statusCode": "200",
+      "statusMessage": "OK",
+      "headers": {
+        "content-type": "application/json",
+        "date": "Wed, 03 Jul 2013 13:30:53 GMT",
+      "server": "gunicorn/0.17.4",
+      "content-length": "30",
+      "connection": "keep-alive"
+      },
+      "body": "{\n  \"origin\": \"94.113.241.2\"\n}"
+    };
+    """
+    And you define following expected HTTP response object:
+    """
+    const expected = {
+      "statusCode": "200",
+      "headers": {
+        "content-type": "application/json",
+        "date": "Wed, 03 Jul 2013 13:30:53 GMT",
+        "server": "gunicorn/0.17.4",
+        "content-length": "30",
+        "connection": "keep-alive"
+      },
+      "body": "{\n  \"origin\": \"94.113.241.2\"\n}",
+    };
+    """
+
+  @stable
+  Scenario: validate
+    When you call:
+    """
+    gavel.validate(expected, real);
```

Comment:
Gavel's API is the same for validating requests and responses. Would it make sense to make that different in the future? They take different structures and it may be clearer to have `validateRequest`, `validateResponse` for the consumer.

Reply:
I've reasoned about making a single function that validates given HTTP message due to the previous implementation. Previously, Gavel had parametric validation of requests and responses:

```js
gavel.validate(real, expected, 'request', callback)
```

However, internally there has been no difference whether you pass `"request"` or `"response"`. It only ensured that a proper `HttpRequest` or `HttpResponse` instance would be created, but they both inherited from `Validatable` class that actually performed the validation.

While this sounds like a good differentiation and better clarity on the usage level, it led to horrible over-engineering internally, creating different class instances that do exactly the same.

Of course, it's an implementation question and can be dealt with gracefully. I do dislike a little giving a customer a false sense of difference between `validateRequest` and `validateResponse`, as the validation performed is exactly the same (validated is what's given). As Gavel doesn't currently perform any input validation (neither in implementation or in spec), it would only confuse people, imo. I can see it beneficial if `validateRequest` would throw when given a response message. Perhaps, we can do this separation in the future together with the input validation, if necessary.

Reply:
Ok, that makes a lot of sense. As I wasn't that familiar with the implementation details I never knew about these details. Thanks for the explanation.